File: Code/Aula_05/Ex7.c
#include <stdio.h>
#include <fcntl.h>
#include <unistd.h>
#include <stdlib.h>

int main(int argc, const char * argv[])
{
	int fp;
	float pi = 3.1415;
   float pilido;

	fp = open ("exercicio7.txt", O_RDWR | O_CREAT, S_IRWXU);
	if(fp==-1)
	{
		printf ("Erro na abertura do arquivo.\n");
		exit (1);
	}
	write(fp, &pi, sizeof(float));
	close(fp);

	fp = open ("exercicio7.txt", O_RDONLY);
	read(fp, &pilido, sizeof(float));
	close(fp);

	printf("\nO valor de PI (lido do arquivo) eh: %f\n\n", pilido);
	return(0);
}

File: 2_Programação_sistemas_operacionais/2.2_File_POSIX/Python/Ex7.py
import os, struct, stat

try:
	fd = os.open("exercicio7.txt", os.O_RDWR | os.O_CREAT, stat.S_IRWXU)
except OSError:
	print("Erro na abertura do arquivo.")
	exit(-1)

pi = float(3.1415)
os.write(fd, struct.pack('f',pi))
os.close(fd)
fd = os.open("exercicio7.txt", os.O_RDONLY)
pilido = struct.unpack('f',os.read(fd,4))
print("O valor de PI (lido do arquivo) é", pilido[0])
os.close(fd)